src/imagestructs.h

```diff
@@ -0,0 +1,28 @@
+#ifndef IMAGESTRUCTS_H
+#define IMAGESTRUCTS_H
+
+#include <string>
+
+// possible aspect ratios of an image
+enum ImageAspect { ImageAspect_Landscape = 0, ImageAspect_Portrait, ImageAspect_Any, ImageAspect_Monitor /* match monitors aspect */ };
+
+// options to consider when displaying an image
+struct ImageDisplayOptions
+{
+    ImageAspect onlyAspect = ImageAspect_Any;
+    bool fitAspectAxisToWindow = false;
+};
+
+// details of a particular image
+struct ImageDetails
```

Comment:
should be a class

Reply:
Unfortunately I've made a LOT of changes in my main branch for this project and backporting the changes here would be some mucking around.
I'm going to open a new PR with these changes and all my new ones, I'll take in your feedback on the above in that PR.
Let me know if you want me to close out this PR.

Reply:
PR is here: https://github.com/NautiluX/slide/pull/46
It contains all the changes in this PR plus lots of new ones. In theory if you approve/commit this PR it should slipstream well into the new one (which has the aspect logic change implemented as you requested).